pysat/_meta.py

```diff
@@ -764,163 +534,387 @@ def match_name(func, name, names):
 
-    @property
-    def min_label(self):
-        return self._min_label
+        if not does_contain:
+            if other.lower() in [i.lower() for i in self.keys_nD()]:
+                does_contain = True
 
-    @property
-    def max_label(self):
-        return self._max_label
+        return does_contain
 
-    @property
-    def fill_label(self):
-        return self._fill_label
+    def __eq__(self, other_meta):
+        """ Check equality between Meta instances
 
-    @units_label.setter
-    def units_label(self, new_label):
-        self._label_setter(new_label, self._units_label, 'units_label', '')
+        Parameters
+        ----------
+        other_meta : Meta
+            A second Meta class object
 
-    @name_label.setter
-    def name_label(self, new_label):
-        self._label_setter(new_label, self._name_label, 'name_label',
-                           use_names_default=True)
+        Returns
+        -------
+        bool
+            True if equal, False if not equal
 
-    @notes_label.setter
-    def notes_label(self, new_label):
-        self._label_setter(new_label, self._notes_label, 'notes_label', '')
+        Note
+        ----
+        Good for testing.
 
-    @desc_label.setter
-    def desc_label(self, new_label):
-        self._label_setter(new_label, self._desc_label, 'desc_label', '')
+        Checks if variable names, attribute names, and metadata values
+        are all equal between to Meta objects. Note that this comparison
+        treats np.NaN == np.NaN as True.
 
-    @plot_label.setter
-    def plot_label(self, new_label):
-        self._label_setter(new_label, self._plot_label, 'plot_label',
-                           use_names_default=True)
+        Name comparison is case-sensitive.
 
-    @axis_label.setter
-    def axis_label(self, new_label):
-        self._label_setter(new_label, self._axis_label, 'axis_label',
-                           use_names_default=True)
+        """
 
-    @scale_label.setter
-    def scale_label(self, new_label):
-        self._label_setter(new_label, self._scale_label, 'scale_label',
-                           'linear')
+        if isinstance(other_meta, Meta):
+            # check first if variables and attributes are the same
+            # quick check on length
+            keys1 = [i for i in self.keys()]
+            keys2 = [i for i in other_meta.keys()]
+            if len(keys1) != len(keys2):
+                return False
 
-    @min_label.setter
-    def min_label(self, new_label):
-        self._label_setter(new_label, self._min_label, 'min_label', np.NaN)
+            # now iterate over each of the keys in the first one
+            # don't need to iterate over second one, if all of the first
+            # in the second we are good. No more or less items in second from
+            # check earlier.
+            for key in keys1:
+                if key not in keys2:
+                    return False
 
-    @max_label.setter
```

Comment:
The labels used to actively update the underlying pandas data object whenever a label was changed by calling the generalized self._label_setter.